### PR TITLE
slightly improve path_alias docs

### PR DESCRIPTION
**Pull request by BenTheElder**
ref: https://github.com/kubernetes/test-infra/issues/9415

/assign @cjwagner @stevekuznetsov

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Oh, I see -- wrong doc. Maybe we can use common language with what we have in GoDoc if you feel like that has an adequate explanation?

**Comment by BenTheElder** [comment]
I'm up for either, but the current docs are actually wrong as they say:
> the default of `/go/src/github.com/org/repo/`

which reads as an absolute path and is not the correct absolute path. I'd like to at least fix this. 

In general I think we should document the defaults in more detail and give plenty of job examples, I think @cjwagner has some PRs related to this already in flight (?)

I can take a look at extending this later this week otherwise.

**Comment by fejta** [comment]
Does it have to check out things to `/home/prow/go` or is this configurable?

**Comment by BenTheElder** [comment]
yep: https://github.com/kubernetes/test-infra/blob/2f64256fa26e76084c2f6b39e1575aa6e43fde18/prow/pod-utils/decorate/podspec.go#L45

(used in many places, doesn't look to be overridable)

File: prow/pod-utilities.md [hunk 1]
```diff
@@ -79,7 +79,7 @@ Additional fields may be required for some use cases:
 	and specify the secret name in the `ssh_key_secrets` field of the job spec.
```
Comment (fejta) [inline prow/pod-utilities.md]: useless comment to remove from my PR dashboard

**Comment by stevekuznetsov** [comment]
/lgtm

@fejta we wanted the user to bring us any image ... in what cases should they ever care what the _absolute_ path is for the _temporary_ files we create in the `EmptyDir` volume we share between `clonerefs` and the test container? If someone has to tailor an image to only allow source code in a specific place for the test infra, we've failed to provide on our promise that you can bring your test container environment and we just magically make it work, right?

**Comment by BenTheElder** [comment]
They might care for legacy tooling that eg depends on a $USER with a $HOME etc. /shrug

An alternate change could look like documenting this only as a relative checkout instead, but I don't think that's particularly helpful or less confusing and you're still probably going to see the full paths in your logs etc anyhow.

**Pull request merged**